pkg/metrics/process/sampler_darwin_test.go

```diff
@@ -153,3 +153,45 @@ func TestProcessSampler_Sample_DockerDecorator(t *testing.T) {
 
 	mock.AssertExpectationsForObjects(t, ctx, harvester)
 }
+
```

Comment:
perhaps adding the case where `hasConfig==false`

Reply:
Good point. There is no real use case or way to call this without config though. I guess that this check was added as `AgentContext` is an Interface and config is a pointer. So this makes it secure from execution pov.

Reply:
Done in 056beb8d1b3c041018fa8a91dba33fb9312bb095

Let me know what do you think :)

Reply:
oh i see, is like there are 2 kinds of default , i got it . I think the change is more correct 👍

Reply:
Me too, thanks for the 👁️ !!